From: Ben Hutchings <ben@decadent.org.uk>
Date: Fri, 23 Oct 2015 00:55:53 +0100
Subject: signal: Fix ABI change in 4.2.4
Forwarded: not-needed

The field signal_struct::group_rwsem was re-added.  This structure is
always allocated by copy_signal() and only used by built-in code, so
it's safe to add fields at the end but not in the middle.  Move it to
the end and hide it from genksyms.

--- a/include/linux/sched.h
+++ b/include/linux/sched.h
@@ -754,6 +754,17 @@ struct signal_struct {
 	unsigned audit_tty_log_passwd;
 	struct tty_audit_buf *tty_audit_buf;
 #endif
+
+	oom_flags_t oom_flags;
+	short oom_score_adj;		/* OOM kill score adjustment */
+	short oom_score_adj_min;	/* OOM kill score adjustment min value.
+					 * Only settable by CAP_SYS_RESOURCE. */
+
+	struct mutex cred_guard_mutex;	/* guard against foreign influences on
+					 * credential calculations
+					 * (notably. ptrace) */
+
+#ifndef __GENKSYMS__
 #ifdef CONFIG_CGROUPS
 	/*
 	 * group_rwsem prevents new tasks from entering the threadgroup and
@@ -766,15 +777,7 @@ struct signal_struct {
 	 */
 	struct rw_semaphore group_rwsem;
 #endif
-
-	oom_flags_t oom_flags;
-	short oom_score_adj;		/* OOM kill score adjustment */
-	short oom_score_adj_min;	/* OOM kill score adjustment min value.
-					 * Only settable by CAP_SYS_RESOURCE. */
-
-	struct mutex cred_guard_mutex;	/* guard against foreign influences on
-					 * credential calculations
-					 * (notably. ptrace) */
+#endif
 };
 
 /*
